code/client/clrcore/MonoScriptRuntime.cs

```diff
@@ -140,7 +140,9 @@ public int GetInstanceId()
 		[SecuritySafeCritical]
 		public int HandlesFile(string filename, IScriptHostWithResourceData metadata)
 		{
-			return (filename.EndsWith(".net.dll") ? 1 : 0);
+			metadata.GetNumResourceMetaData("mono_rt2", out IntPtr enableV2);
```

Comment:
FWIW eventually I'd like to see this be detected using assembly metadata, somehow.